lib/prime.rb

```diff
@@ -132,7 +209,8 @@ def method_added(method) # :nodoc:
   #   Upper bound of prime numbers. The iterator stops after it
   #   yields all prime numbers p <= +ubound+.
   #
-  def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  #def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  def each(ubound = nil, generator = NextPrimeGenerator.new, &block)
```

Comment:
I am not that familiar with prime numbers, so I can only hope to help with the implementation.

Your default ``prime?`` could be simplified, but wouldn't make such a great impact.
```Ruby
def prime?
  return true if [2, 3, 5, 7].include? self
  return false unless self > 1 and 210.gcd(self % 210) == 1
  Prime::A014233.each do |ix, mx| # Remove pair, use (ix, mx) directly
    return false unless miller_rabin_test(ix)
    break if self < mx
  end
  self == 41 || miller_rabin_test(41) # Simplify return value
end
```

With your ``prime1?`` and ``prime2?`` there is more room to make a significant change.
There is a typo in ``prime2``, comma instead of dot for ``val = Prime::A014233.values.sort,detect {|v| v > self}``.
```Ruby
def prime1?
  return true if [2, 3, 5, 7].include? self
  return false unless self > 1 and 210.gcd(self % 210) == 1
  witnesses = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41]
  # Isn't Prime::A014233.values already returning in the defined order in recent Ruby versions?
  # Otherwise Prime::A014233.values.sort! could be stored in a constant
  # array.sort! can be used here
  # Some values are tested twice as they repeat in Prime::A014233,
  # perhaps a constant could help here
  val = Prime::A014233.values.sort!.detect {|v| v > self}
  # Prime::A014233.key(val) is a constant inside the iterator, calculate it once outside
  if val
    # Since hash.key is used we could flip key/values to take more advantage of the Hash,
    # but then keys would collide, are repeated values needed?
    # Prime::A014233.key(3_825_123_056_546_413_051) should return one of [23,29,31] or 23 is enough for example?
    val = Prime::A014233.key(val)
    witnesses.select! { |p| p <= val }
  end
  # Simplify return value for the serial version, return false if any miller rabin test fails, otherwise true
  witnesses.all? { |p| miller_rabin_test(p) }
end
```